spec/outputs/elasticsearch_spec.rb

```diff
@@ -306,6 +306,56 @@
     end
   end
 
+  describe "http client create" do
+    require "logstash/outputs/elasticsearch"
+    require "elasticsearch"
+
+    let(:options) {
+      {
+        "protocol" => "http",
+        "index" => "my-index",
+        "host" => "localhost",
+        "path" => "some-path"
+      }
+    }
+
+    let(:eso) {LogStash::Outputs::ElasticSearch.new(options)}
+
+    let(:manticore_host) {
+      eso.client.first.send(:client).transport.options[:host].first
+    }
+
+    around(:each) do |block|
+      thread = eso.register
```

Comment:
register shouldn't return anything, if it does, it's not expected.

The register method for this plugin is already pretty wild, I'll make a note that it needs some love to make it easier to manage and test.